src/cli.ts

```diff
@@ -76,28 +76,56 @@ Provide it via the ${AUTOBLOCKS_API_KEY_NAME} environment variable, within a .en
 
 You can get your API key from ${AUTOBLOCKS_WEBAPP_BASE_URL}/settings/api-keys`;
 
+const commandNotFoundErrorMessage = `
+No command found. Provide a command following '--'. For example:
+
+npx autoblocks testing exec -- echo "Hello, world!
+`;
+
+const parseCommandFromArgv = (
+  argv: yargs.Arguments,
+): { command: string; commandArgs: string[] } => {
+  const unparsed = argv['--'];
+  if (!Array.isArray(unparsed)) {
+    throw new Error(commandNotFoundErrorMessage);
+  }
+
+  const [command, ...commandArgs] = unparsed.map(String);
+
+  if (!command) {
+    throw new Error(commandNotFoundErrorMessage);
+  }
+
+  return { command, commandArgs };
+};
+
+const apiKeyOptions = {
+  describe: `Autoblocks API key. Can be set via the ${AUTOBLOCKS_API_KEY_NAME} environment variable.`,
+  type: 'string',
+  default: variableFromEnv(AUTOBLOCKS_API_KEY_NAME),
+  demandOption: apiKeyMissingErrorMessage,
+} as const;
+
+const portOptions = {
+  alias: 'p',
+  describe: 'Local CLI server port number',
+  type: 'number',
+  default: 5555,
+} as const;
```

Comment:
For `as const` type assertion - 

Is this shorthand for doing something like
```
const portOptions: { alias: string, describe: string, type: string, default: number } = {}
```

Assuming this is so below where you are using these values it doesn't complain about these just being plain objects?

Reply:
`as const` gives the object the narrowest type and makes it readonly. in this case the relevant part is that instead of `type: string` we get `type: 'string'`, since yargs options expect a union type of `'number' | 'string' | etc` instead of a string. so if you drop `as const` you'd get a type error:

<img width="911" alt="Screenshot 2024-04-10 at 2 21 43 PM" src="https://github.com/autoblocksai/cli/assets/7498009/311d8394-88fd-46bd-9ee9-2c348ef1ab78">

so the type without `as const` is:

<img width="243" alt="Screenshot 2024-04-10 at 2 23 04 PM" src="https://github.com/autoblocksai/cli/assets/7498009/11ea1068-4573-46d6-8546-ee325e25fc38">

and with it is:

<img width="902" alt="Screenshot 2024-04-10 at 2 23 11 PM" src="https://github.com/autoblocksai/cli/assets/7498009/558c16cc-0f21-4d1f-83b1-acf6a5037bc6">

Reply:
Great explanation - thanks!